src/backend/columnar/columnar_storage.c

```diff
@@ -347,10 +347,34 @@ ColumnarStorageIsCurrent(Relation rel)
 
 
 /*
- * ColumnarStorageReserveStripe - reserve stripe ID and row numbers.
+ * ColumnarStorageReserveRowNumber returns reservedRowNumber and advances
+ * it for next row number reservation.
  */
 uint64
-ColumnarStorageReserveStripe(Relation rel, uint64 nrows, uint64 *firstRowNumber)
+ColumnarStorageReserveRowNumber(Relation rel, uint64 nrows)
+{
+	LockRelationForExtension(rel, ExclusiveLock);
+
+	ColumnarMetapage metapage = ColumnarMetapageRead(rel, false);
+
+	uint64 firstRowNumber = metapage.reservedRowNumber;
+	metapage.reservedRowNumber += nrows;
+
+	WriteToBlock(rel, COLUMNAR_METAPAGE_BLOCKNO, SizeOfPageHeaderData,
+				 (char *) &metapage, sizeof(ColumnarMetapage),
+				 true /* clear because we are overwriting */);
```

Comment:
maybe we can have a function for that block (e.g: `ColumnarOverwriteMetapage`) and we can use it in `ColumnarStorageReserveStripe` too